src/main/java/org/bonitasoft/connectors/scripting/ShellConnector.java

```diff
@@ -134,60 +130,46 @@ else if (interpreter.contains("sh"))
             return ".sh";
         else
             return "";
-    
     }
 
     private Process runScript(final String interpreterInput, final String parameterInput, File script) throws IOException {
         String args[] = { interpreterInput, parameterInput, script.getCanonicalPath() };
         Process process = null;
 		try {
-			process = Runtime.getRuntime().exec(args);
-		} catch (IOException e) {
-			e.printStackTrace();
+            process = new ProcessBuilder(args).redirectErrorStream(true).start();
+            return process;
 		} finally{
 			if(process != null){
-				// close unused streams
-				try {
-					process.getOutputStream().close();
-				} catch (IOException e) {
-					LOGGER.warning("Unable to close " + e.getMessage());
-				}
-				try {
-					process.getErrorStream().close();
-				} catch (IOException e) {
-					LOGGER.warning("Unable to close " + e.getMessage());
-				}			
+                closeQuietly(process.getOutputStream());
+                closeQuietly(process.getErrorStream());
```

Comment:
hm, the problem is that the line between my change use tabs instead of spaces, that's why the GitHub rendering show
a kind of formatting issue. I only reformat the part of the code I changed to avoid introducing too much modifications

I will manage this in another PR